src/structures/idt.rs

```diff
@@ -1051,7 +1051,19 @@ impl SelectorErrorCode {
 
     /// The index of the selector which caused the error.
     pub fn index(&self) -> u64 {
-        self.flags.get_bits(3..16)
+        let descriptor_table = match self.flags.get_bits(1..3) {
+            0b00 => DescriptorTable::Gdt,
+            0b01 => DescriptorTable::Idt,
+            0b10 => DescriptorTable::Ldt,
+            0b11 => DescriptorTable::Idt,
+            _ => unreachable!(),
+        };
```

Comment:
This can be replaced by the `descriptor_table` getter which contains pretty much the same code.

Reply:
Not sure if it'll pass CI, but just submitted a version that calls said getter instead of trying to rewrite that match statement.